apps/dapp/src/components/Navbar/NavDropdown.tsx

```diff
@@ -1,28 +1,43 @@
 import { useGetProfileData } from '@/hooks/useGetProfileData';
 import { useHasAccountProfile } from '@/hooks/useHasAccountProfile';
+import { toast } from '@/hooks/useToast';
 import { ChevronDownIcon } from '@/ui/Icons';
 import {
   DropdownMenu,
   DropdownMenuContent,
   DropdownMenuTrigger,
 } from '@/ui/UiDropdownMenu';
-
+import { useEffect } from 'react';
 import { useAccount } from 'wagmi';
 import { UserProfileBadge } from '../Profile/UserProfileBadge';
 import DisconnectWallet from '../Wallet/DisconnectWallet';
 import UserLinks from './UserLinks';
 
 export const NavBarDropdown = () => {
-  const { address, isConnected } = useAccount();
+  const { isConnected } = useAccount();
   const { hasProfile } = useHasAccountProfile();
-  const { profileData } = useGetProfileData(address);
-  const { name, image } = profileData;
+  const { profileData, isLoading, getProfileDataError } = useGetProfileData();
+
+  useEffect(() => {
+    if (getProfileDataError) {
+      toast({
+        title: 'Error fetching profile data',
+        description: getProfileDataError?.message,
+        variant: 'error',
+      });
+    }
+  }, [getProfileDataError]);
 
   return (
     <DropdownMenu>
       <DropdownMenuTrigger className="focus:outline-none">
         <div className="flex gap-2 items-center">
-          {hasProfile && <UserProfileBadge name={name} imageUrl={image} />}
+          {hasProfile && profileData && (
```

Comment:
is `hasProfile` necessary? `profileData` should be valid data or undefined. Less variables to deal with that way

Reply:
removed